src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -218,7 +218,7 @@ export class TerminalService implements ITerminalService {
 							this.splitInstance(terminalInstance, { attachPersistentTerminal: terminalLayout.terminal! });
 						}
 					});
-					const activeInstance = this.terminalInstances.find(t => t.shellLaunchConfig.attachPersistentTerminal?.pid === tabLayout.activeTerminalProcessId);
+					const activeInstance = this.terminalInstances.find(t => t.persistentTerminalId === tabLayout.activeTerminalProcessId);
```

Comment:
This seems to be confusing pid with persistent terminal id, I'll look through everything and fix this up